core/drivers/wdt/watchdog_sm.c

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 Microchip
+ */
+
+#include <drivers/wdt.h>
+#include <sm/optee_smc.h>
+#include <sm/psci.h>
+
+enum smcwd_call {
+	SMCWD_INIT		= 0,
+	SMCWD_SET_TIMEOUT	= 1,
+	SMCWD_ENABLE		= 2,
+	SMCWD_PET		= 3,
+	SMCWD_GET_TIMELEFT	= 4,
+};
+
+static unsigned long wdt_min_timeout;
+static unsigned long wdt_max_timeout;
+
+enum sm_handler_ret wdt_sm_handler(struct thread_smc_args *args)
```

Comment:
I'd like to see a definition of the ABI provided by this function. From what I've been able to find so it seems that this is specific for mt8173.

Reply:
Actually, the definition does not seems to be specified by ARM. But the driver that was submitted states that is a generic SMC watchdog driver [1]. If not, were should I place this file ?

[1] https://lore.kernel.org/linux-arm-kernel/20200220155159.GB29658@roeck-us.net/T/

Reply:
I assume that this might be reused for other platforms. Let's keep it here, but we should be clear on what we're defining.